### PR TITLE
Fix snapshot in data offline GTs and Egamma regression, and update Strip 2018 MC conditions

### PR DESCRIPTION
**Pull request by tocheng**
#### PR description:

This PR contains a fix of snapshot for data offline GTs addressed at [1].
EGamma PF cluster correction tags[2] for 2018 ultra-legacy reprocessing are updated due a bug described at [3]. The corrections affect 2018 MC and the 2018 IOVs of the offline GTs. Since the same corrections used for 2018 are intended to be used for Run 3 as well, the Run 3 MC GTs are updated as well. 
Strip MC conditions for 2018 are updated to have a better data V.S. MC comparison[4]

[1] https://hypernews.cern.ch/HyperNews/CMS/get/calibrations/4223.html
[2] https://hypernews.cern.ch/HyperNews/CMS/get/calibrations/4222/1/1.html
[3] https://indico.cern.ch/event/842863/contributions/3562026/attachments/1906927/3149650/updates_2018ULPFCorrections.pdf
[4]https://indico.cern.ch/event/854616/#7-strip-mc-conditions-for-ul-2

GT diffs:
**Data offline GTs**
https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/106X_dataRun2_v24/106X_dataRun2_v23
https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/106X_dataRun2_relval_v22/106X_dataRun2_relval_v21

**Data prompt-like GTs**
https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/106X_dataRun2_PromptLike_v13/106X_dataRun2_PromptLike_v12
https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/106X_dataRun2_PromptLike_HEfail_v12/106X_dataRun2_PromptLike_HEfail_v11
https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/106X_dataRun2_PromptLike_HI_v13/106X_dataRun2_PromptLike_HI_v12

**2018 MC GTs**
https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/106X_upgrade2018_design_v6/106X_upgrade2018_design_v5
https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/106X_upgrade2018_realistic_v11/106X_upgrade2018_realistic_v8
https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/106X_upgrade2018_realistic_HEfail_v11/106X_upgrade2018_realistic_HEfail_v8
https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/106X_upgrade2018_realistic_HI_v6/106X_upgrade2018_realistic_HI_v5
https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/106X_upgrade2018cosmics_realistic_deco_v7/106X_upgrade2018cosmics_realistic_deco_v4
https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/106X_upgrade2018cosmics_realistic_peak_v8/106X_upgrade2018cosmics_realistic_peak_v4

**Run 3 MC GTs**
https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/106X_upgrade2021_design_v5/106X_upgrade2021_design_v4
https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/106X_upgrade2021_realistic_v11/106X_upgrade2021_realistic_v10
https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/106X_upgrade2021cosmics_realistic_deco_v4/106X_upgrade2021cosmics_realistic_deco_v3

#### PR validation:
See the description above for the physics validation of the tags. 

In addition, a technical test was performed: runTheMatrix.py -l limited -i all --ibeos

### CODE REVIEW

**Comment by tocheng** [comment]
@cmsbuild please test

**Comment by srimanob** [comment]
We are really waiting for the new release for final validation. If it's not ready, I think I would like to propose to make the release now, and this goes to the next one.

**Comment by srimanob** [comment]
@mmusich 
The GT will be used for sure. But the autocond can wait if the PR is not ready.

**Comment by mmusich** [comment]
> The GT will be used for sure. 

Thanks Phat.

**Comment by boudoul** [comment]
Hi @tocheng  what about run 3 data ?

**Comment by christopheralanwest** [comment]
@boudoul @tocheng Since the EGamma regression is not needed for cosmic data, I would prefer to wait until the 2018 conditions are validated before updating the Run 3 prompt GT.

**Comment by tocheng** [comment]
+1

**Comment by tocheng** [comment]
@cmsbuild please test

**Comment by Martin-Grunewald** [comment]
Looks like the older L1T menu is not compatible with the HLT menu inside the 10_6 release and used by RelVals.

**Comment by srimanob** [comment]
@Martin-Grunewald 
Yeah, this is what @Sam-Harper expects when we decided to change to old L1 menu, to match with UL. The pure relvals (without 10_2 UL) will face this issue, as 2018 is used, not fake menu.

Should we keep v9 here, or we change the HLT menu to match with this old L1?

**Comment by Martin-Grunewald** [comment]
If possible, I'd prefer not to downgrade the HLT menu in 10_6. The following options come to mind:
Can UL production just use the new GT on the cmsDriver cmdline, rather than putting it into autoCond.py?
Can UL production overwrite just the L1T menu on the cmsDriver cmdline, with the GTs updated except for downgrading the L1T menu?

**Comment by srimanob** [comment]
Discussed with Alca, we will have 
+ 106X_upgrade2018_realistic_v11 with L1 (v2), 
+ and 106X_upgrade2018_realistic_v11_L1v1 (name may difference, with L1 (v1). 

 106X_upgrade2018_realistic_v11 will be used as phase1_2018_realistic

**Comment by Sam-Harper** [comment]
Sounds good, thanks! I like that its clear this is L1 v1

**Comment by Martin-Grunewald** [comment]
Hmm, as far as I can see you downgraded the L1T menu in 5 GTs in autoCond, so that should be reverted in all 5 GTs in autoCond.py as to avoid confusion, and only the special L1v1 GT has the downgrated L1T menu...

**Comment by tocheng** [comment]
As we discussed, we revert the L1 menu in GTs. We will make new set of 2018 GTs to include new strip MC conditions.

**Comment by fabiocos** [comment]
please test

**Comment by tocheng** [comment]
@fabiocos I will submit a new commit for 2018 simulation GTs to include new strip MC conditions this evening.

**Comment by tocheng** [comment]
@cmsbuild please test

**Comment by tocheng** [comment]
+1

**Comment by mmusich** [comment]
Hi @tocheng, regarding https://github.com/cms-sw/cmssw/commit/e859edc7081e3d4dc78e4381938e40d34ec9a07f (thanks!) does a corresponding update exist in master as well?

**Comment by tocheng** [comment]
> Hi @tocheng, regarding [e859edc](https://github.com/cms-sw/cmssw/commit/e859edc7081e3d4dc78e4381938e40d34ec9a07f) (thanks!) does a corresponding update exist in master as well?

@mmusich No. We hope this PR #28134 can be merged first. Then we can update strip MC conditions for 2018.

**Comment by mmusich** [comment]
> We hope this PR #28134 can be merged first. Then we can update strip MC conditions for 2018.

Ok. Thanks. I think PdmV is going to provide specific relvals for UL in 10.6.x but would be nice to pass the update also through the regular release validation for next 11.0.x pre-release.

File: Configuration/AlCa/python/autoCond.py [hunk 1]
```diff
@@ -52,23 +52,23 @@
...
+    'phase1_2018_realistic_hi' :  '106X_upgrade2018_realistic_HI_v6',
     # GlobalTag for MC production with realistic conditions for full Phase1 2018 detector: HEM-15-16 fail
-    'phase1_2018_realistic_HEfail' :  '106X_upgrade2018_realistic_HEfail_v8',
+    'phase1_2018_realistic_HEfail' :  '106X_upgrade2018_realistic_HEfail_v11',
     # GlobalTag for MC production (cosmics) with realistic conditions for full Phase1 2018 detector,  Strip tracker in DECO mode
-    'phase1_2018_cosmics'      :   '106X_upgrade2018cosmics_realistic_deco_v4',
+    'phase1_2018_cosmics'      :   '106X_upgrade2018cosmics_realistic_deco_v7',
     # GlobalTag for MC production (cosmics) with realistic conditions for full Phase1 2018 detector,  Strip tracker in PEAK mode
-    'phase1_2018_cosmics_peak' :   '106X_upgrade2018cosmics_realistic_peak_v4',
+    'phase1_2018_cosmics_peak' :   '106X_upgrade2018cosmics_realistic_peak_v7',
```
Comment (mmusich) [inline Configuration/AlCa/python/autoCond.py]: @tocheng @christopheralanwest 
```
$ conddb diff 106X_upgrade2018cosmics_realistic_peak_v7 106X_upgrade2018cosmics_realistic_peak_v4 | grep SiStrip
SiStripApvGain2Rcd    -                                                 SiStripApvGain_FromParticles_Realistic2018MC_v0_mc  SiStripApvGain_FromParticles_DecoMode_v6_mc           
SiStripApvGainRcd     -                                                 SiStripApvGain_Realistic2018MC_v0_mc                SiStripApvGain_RealisticMC_v2_mc                      
SiStripApvGainSimRcd  -                                                 SiStripApvGain_Realistic2018MCSim_v0_mc             SiStripApvGain_RealisticMCSim_v3_mc                   
SiStripNoisesRcd      -                                                 SiStripNoise_DecoMode_2018_Minus20C_v0_mc           SiStripNoise_PeakMode_TickCorr_Minus15C_Fixed_v2_mc   
```
as we've discussed via e-mail `SiStripNoise_DecoMode_2018_Minus20C_v0_mc` can't be used in the peak GT, please use `SiStripNoise_PeakMode_2018_Minus20C_v0_mc` instead.

**Comment by mmusich** [comment]
@tocheng 

> We hope this PR #28134 can be merged first.

I see that #28134 is merged. What's the plan to:
   * update the data offline GT snapshot also for master?
   * introduce the new SiStrip tags for 2018 simulation also for master? 

Thanks.

**Comment by christopheralanwest** [comment]
-1

As discussed at https://github.com/cms-sw/cmssw/pull/28053#discussion_r337487241, the cosmic peak mode GT needs to be updated.

**Comment by christopheralanwest** [comment]
> @tocheng
> 
> > We hope this PR #28134 can be merged first.
> 
> I see that #28134 is merged. What's the plan to:
> 
>     * update the data offline GT snapshot also for master?
> 
>     * introduce the new SiStrip tags for 2018 simulation also for master?
> 
> 
> Thanks.

PR #28134 updated the snapshot time for the data GT in the autoCond of master as well. See the IOV differences in the [GT diff](https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/110X_dataRun2_v8/110X_dataRun2_v7). The PR to master that updates the SiStrip conditions is #28234.

**Comment by tocheng** [comment]
@cmsbuild please test

**Comment by tocheng** [comment]
@cmsbuild please test workflow 7.2,7.4

**Comment by tocheng** [comment]
> @tocheng
> 
> > We hope this PR #28134 can be merged first.
> 
> I see that #28134 is merged. What's the plan to:
> 
>     * update the data offline GT snapshot also for master?
> 
>     * introduce the new SiStrip tags for 2018 simulation also for master?
> 
> 
> Thanks.

Hello @mmusich , the last commit adds the peak mode noise to the GT.

**Comment by tocheng** [comment]
+1
2018UL validation reported at PPD general meetings
https://indico.cern.ch/event/853103/
https://indico.cern.ch/event/857175/
The 2018 UL conditions are merged into master.

**Comment by fabiocos** [comment]
+1

**Pull request merged**